dotnet-interactive/JupyterKernelSpecInstaller.cs

```diff
@@ -2,77 +2,155 @@
+            if (succeeded)
+            {
+                _console.Out.WriteLine($"Installed \"{kernelDisplayName}\" kernel.");
             }
+            else
+            {
+                _console.Error.WriteLine(
+                    $"Failed to install \"{kernelDisplayName}\" kernel.");
+            }
+
+            return succeeded;
         }
 
-        public async Task<KernelSpecInstallResult> UninstallKernel(DirectoryInfo sourceDirectory)
+        private string GetKernelDisplayName(DirectoryInfo directory)
         {
-            var commandLineResult = await _kernelSpecModule.UninstallKernel(sourceDirectory);
-            var message = string.Join('\n', commandLineResult.Output.Concat(commandLineResult.Error));
+            var kernelSpec = directory.GetFiles("kernel.json", SearchOption.AllDirectories).Single();
 
-            var result = new KernelSpecInstallResult(commandLineResult.ExitCode == 0,
-                message
-            );
+            var parsed = JObject.Parse(File.ReadAllText(kernelSpec.FullName));
+            return parsed["display_name"].Value<string>();
+        }
 
-            return result;
+        public async Task<bool> UninstallKernel(string kernelspecName)
+        {
+            if (string.IsNullOrWhiteSpace(kernelspecName))
+            {
+                throw new ArgumentException("Value cannot be null or whitespace.", nameof(kernelspecName));
+            }
+
+            try
+            {
+                var result = await _kernelSpecModule.UninstallKernel(kernelspecName);
+                if (result.ExitCode == 0)
+                {
+                    _console.Out.WriteLine($"Uninstalled \"{kernelspecName}\" kernelspec.");
+                    return true;
+                }
+            }
+            catch (Win32Exception w32e)
+            {
+                // file not found when executing process
+                if (!w32e.Source.Contains(typeof(System.Diagnostics.Process).FullName))
+                {
+                    _console.Error.WriteLine($"Failed installing \"{kernelspecName}\" kernelspec.");
+                    throw;
+                }
+            }
+
+            var location = new DirectoryInfo(Path.Combine(_kernelSpecModule.GetDefaultKernelSpecDirectory().FullName, kernelspecName));
+
+            _console.Out.WriteLine("The kernelspec module is not available.");
+            if (!location.Exists)
+            {
+                _console.Error.WriteLine($"Directory {location.FullName} does not exists");
+                return false;
+            }
+
+            try
+            {
+                var kernelDisplayName = GetKernelDisplayName(location);
+
+                location.Delete(true);
+                _console.Out.WriteLine($"Uninstalled \"{kernelDisplayName}\" kernel.");
+
+            }
+            catch (IOException ioe)
+            {
+                _console.Error.WriteLine(ioe.Message);
```

Comment:
What is an example of this message? Is it useful information for the user?

Reply:
Done